tests/CMakeLists.txt

```diff
@@ -5,6 +5,11 @@ configure_file(
 
 add_executable(frozen.tests "")
 target_link_libraries(frozen.tests PUBLIC frozen::frozen)
+
+if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+  target_compile_features(frozen INTERFACE cxx_std_17)
```

Comment:
The interface C++ version should be the minimum required, because as of written it is now a requirement of the library to use C++17. I would instead put that on targets that need C++17 such as tests.

Reply:
@gracicot: I leave it to you to decide if current CMake implementation is ok, you seem to have a better command of cmake then I do :-)